components/eamxx/src/physics/rrtmgp/rrtmgp_utils.hpp

```diff
@@ -98,28 +99,31 @@ bool check_range(T x, Real xmin, Real xmax, std::string msg, std::ostream& out=s
         }
       });
     auto num_bad = sum(bad_mask);
-    pass = false;
-    out << msg << ": "
-        << num_bad << " values outside range "
-        << "[" << xmin << "," << xmax << "]"
-        << "; minval = " << _xmin
-        << "; maxval = " << _xmax << "\n";
+    if (num_bad > 0) {
+      pass = false;
+      out << msg << ": "
+          << num_bad << " values outside range "
+          << "[" << xmin << "," << xmax << "]"
+          << "; minval = " << _xmin
+          << "; maxval = " << _xmax << "\n";
+    }
   }
   return pass;
 }
 #endif
 #ifdef RRTMGP_ENABLE_KOKKOS
-template <class T>
-bool check_range_k(T x, Real xmin, Real xmax, std::string msg, std::ostream& out=std::cout) {
+template <class T, typename std::enable_if<T::rank == 1>::type* = nullptr>
+bool check_range_k(T x, typename T::const_value_type xmin, typename T::const_value_type xmax,
+                   std::string msg, std::ostream& out=std::cout) {
   bool pass = true;
   auto _xmin = conv::minval(x);
   auto _xmax = conv::maxval(x);
   if (_xmin < xmin or _xmax > xmax) {
     // How many outside range?
-    bool1dk bad_mask("bad_mask", x.size());
-    Kokkos::parallel_for(x.size(), KOKKOS_LAMBDA (int i) {
-      if (x.data()[i] < xmin or x.data()[i] > xmax) {
```

Comment:
When you have subviews, it's no longer safe to use data(). If you do, you'll end up iterating over items that aren't in the subview.

Reply:
That's fine, but does this still check over all elements in the subview? That was the only reason to use data before, to flatten the view and make it easy to check all values (i.e., over ncol, nlay, nbnd, etc).

Reply:
The use of `extent(0)` makes me think not, but this is a little out of my wheelhouse.

Reply:
Okay, it looks like you implemented separate templates for rank = 1, 2, 3 below. Is there a way to avoid this? Seems like a lot of duplicated code.

Reply:
The specializations are needed since we are no longer using the .data() approach that allowed us to treat all ranks of views the same. .data() assumes the view is contiguous and that is not true for some kinds of subviews.